do-not-commit/dict-modify-iterating.py

```diff
@@ -0,0 +1,19 @@
+d = {"a": 1, "b": 2}
+# ruleid:dict-del-while-iterate
+for k, v in d.items():
+    del d[k]
+
+d = {"a": 1, "b": 2}
+# ruleid:dict-del-while-iterate
+for k in d.keys():
+    del d[k]
+
+# ruleid:dict-del-while-iterate
+for k in d.keys():
```

Comment:
It appears that `d[k]` is a dict with items being deleted while in a for loop. This is usually a bad idea and will likely lead to a RuntimeError: dictionary changed size during iteration ([.dict-modify-iterating.dict-del-while-iterate](https://semgrep.dev/r/.dict-modify-iterating.dict-del-while-iterate))
#
<div>
<a href="https://semgrep.dev">
<img align="right" src="https://r2c.dev/semgrep_icon_text.svg" height=28px alt="" />
</a>

<sub> [How do I suppress this?](https://semgrep.dev/docs/running-rules/#managing-findings) **·** [Report a bug](https://github.com/returntocorp/semgrep-rules/issues/new?assignees=&labels=&template=rule_bug_report.md&title=.dict-modify-iterating.dict-del-while-iterate) **·** [Manage rule](https://semgrep.dev/manage/projects)</sub>
</div>